src/test/java/com/rackspace/ceres/app/web/WriteControllerTest.java

```diff
@@ -50,16 +71,39 @@ public void testPutMetrics_WithParams() {
 
     //Tests for details and summary both true
     webTestClient.post().uri(uriBuilder -> uriBuilder.path("/api/put").queryParam("details", true)
-        .queryParam("summary", true).build()).body(Flux.just(metric), Metric.class)
+        .queryParam("summary", true).build()).bodyValue(List.of(metric))
+        .header("X-Tenant", "t-1")
         .exchange()
         .expectStatus().isOk().expectBody(PutResponse.class)
         .isEqualTo(new PutResponse().setSuccess(1).setFailed(0).setErrors(List.of()));
+    verify(dataWriteService).ingest(metrics.capture());
+
+    //verify tenant is same as present in header
+    StepVerifier.create(metrics.getValue()).expectNext(Tuples.of("t-1", metric))
+        .expectComplete().verify();
+
+    verifyNoMoreInteractions(dataWriteService);
+  }
+
+  @Test
+  public  void testPutMetric_MissingDetailsParam() {
```

Comment:
nit: as worded it sounds like it's a "negative" test case
```suggestion
  public  void testPutMetric_JustSummaryParam() {
```